Task5_1/SparseMatrix.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections;
+
+namespace Task5_1
+{
+    internal class SparseMatrix : IEnumerable<long>
+    {
+        private Dictionary<(int, int), long> _sparseMatrix;
+        public int Rows { get; }
+        public int Columns { get; }
+
+        public SparseMatrix(int rows, int columns)
+        {
+            if (rows <= 0 || columns <= 0)
+                throw new ArgumentException("Indices must be greater than zero");
+
+            Rows = rows;
+            Columns = columns;
+            _sparseMatrix = new Dictionary<(int, int), long>();
+        }
+
+        public long this[int i, int j]
+        {
+            get
+            {
+                if (i < 0 || i >= Rows || j < 0 || j >= Columns)
+                    throw new IndexOutOfRangeException();
+
+                return _sparseMatrix.TryGetValue((i, j), out long value) ? value : 0;
+            }
+            set
+            {
+                if (i < 0 || i >= Rows || j < 0 || j >= Columns)
+                    throw new IndexOutOfRangeException();
+
+                if(value != 0)
+                    _sparseMatrix[(i, j)] = value;
+                else
+                    _sparseMatrix.Remove((i, j));
+            }
+        }
+
```

Comment:
The matrix  can be very big, better to use not strings concatenation but  StringBuilder (will be explained in lectures), just for information